core/services/relay/evm/dsl.go

```diff
@@ -0,0 +1,96 @@
+package evm
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+)
+
+type EventBySigFilter struct {
+	Address  common.Address
+	EventSig common.Hash
+}
+
+func NewEventBySigFilter(address common.Address, eventSig common.Hash) query.Expression {
+	var searchEventFilter *EventBySigFilter
+	searchEventFilter.Address = address
+	searchEventFilter.EventSig = eventSig
+	return query.Expression{Primitive: searchEventFilter}
+}
+
+func (f *EventBySigFilter) Accept(visitor primitives.Visitor) {
+	switch v := visitor.(type) {
+	case *PgDSLParser:
+		v.VisitEventBySigFilter(f)
+	}
+}
+
+type EventByTopicFilter struct {
+	EventSig         common.Hash
+	Topic            uint64
+	ValueComparators []primitives.ValueComparator
+}
+
+func NewEventByTopicFilter(eventSig common.Hash, topicIndex uint64, valueComparators []primitives.ValueComparator) query.Expression {
```

Comment:
Does it make sense to have eventSig there? At this moment, all LogPoller queries are always filtered by `(evmChainId, address, eventSig)`. Seems like either `NewEventByTopicFilter` should have address field included or should not have eventSig and keep only filtering by topic, right? What if I want to build query that filters by both `topicFilter` and `wordFilter` (or multiple `topicFilters`)? I will have to put eventSig multiple times which might be very cumbersome from the client perspective

Reply:
I think building existing LP queries using these primitives could be helpful to understand better which abstraction is the most accurate here

Reply:
Good callout. We ended up not needing `eventSig` here as the event signature and address are being passed from the chain reader event binding to the parser. Therefore no event signature or address need to exist in the query. Instead, the binding is identified by the contract name and event name in `QueryKey`.

Reply:
@EasterTheBunny Can you link this change since its in a separate PR